zulipterminal/ui_tools/views.py

```diff
@@ -1041,16 +1041,32 @@ def make_table_with_categories(
             if category:
                 if len(widgets) > 0:  # Separate categories with newline.
                     widgets.append(urwid.Text(""))
-                widgets.append(urwid.Text(("popup_category", category)))
+                if isinstance(category, tuple):
+                    content.append(category)
+                else:
+                    widgets.append(urwid.Text(("popup_category", category)))
             for index, row in enumerate(content):
                 if isinstance(row, str) and row:
                     widgets.append(urwid.Text(row))
                 elif isinstance(row, tuple):
                     label, data = row
-                    strip = urwid.Columns(
-                        [(column_widths[0], urwid.Text(label)), urwid.Text(data)],
-                        dividechars=dividechars,
-                    )
+                    if len(category) > 1 and label == category[0]:
+                        index += 1
+                        strip = urwid.Columns(
+                            [
+                                (
+                                    column_widths[0],
+                                    urwid.Text(("popup_category", label)),
+                                ),
+                                urwid.Text(("popup_category", f"{data} "), "right"),
+                            ],
+                            dividechars=dividechars,
+                        )
+                    else:
+                        strip = urwid.Columns(
+                            [(column_widths[0], urwid.Text(label)), urwid.Text(data)],
+                            dividechars=dividechars,
+                        )
                     widgets.append(
                         urwid.AttrWrap(strip, None if index % 2 else "popup_contrast")
                     )
```

Comment:
Yes updating the loop index is confusing. Figured out a better way to get what I expected.

No we don't need to wrap the popup_contrast. I did it because I was trying to make it visually discernible, updated it to work without the wrap.

We can have a discussion about the style.

Reply:
We don't necessarily need to think about a style in this first version, but at the time it struck me as an additional motivator for adjusting the loop.